File: Java/프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        int time = switchSec(pos);
        int len = switchSec(video_len);
        int opStart = switchSec(op_start);
        int opEnd = switchSec(op_end);
        time = opSkip(time, opStart, opEnd);
        for(int i=0; i<commands.length; i++) {
            switch(commands[i]) {
                case "prev":
                time = prev(time, len, opStart, opEnd);
                break;
                case "next":
                time = next(time, len, opStart, opEnd);
                break;
            }
        }
        return switchTimeFormat(time);
    }
    public int switchSec(String time) {
        int x =  Integer.parseInt(time.replaceAll(":",""));
        return x/100*60 + x%100;
    }
    public String switchTimeFormat(int time) {
        int min = time/60;
        int sec = time%60;
        
        String stringMin = min < 10? "0"+min : min+"";
        String stringSec = sec < 10? "0"+sec : sec+"";
        return stringMin+":"+stringSec;
    }
    public int next(int time, int video_len, int op_start, int op_end) {
        int pos = time+10 > video_len? video_len : time+10;
        return opSkip(pos, op_start, op_end);
    }
    public int prev(int time, int video_len, int op_start, int op_end) {
        int pos = time-10 < 0? 0 : time-10;
        return opSkip(pos, op_start, op_end);
    }
    public int opSkip(int time, int op_start, int op_end) {
        int pos = time;
        if(time >= op_start && time <= op_end) {
            pos=op_end;
        }
        return pos;
    }
}

File: Kotlin/프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.kt
class Solution {
    fun solution(video_len: String, pos: String, op_start: String, op_end: String, commands: Array<String>): String {

        var video_len_second = convertSecond(video_len)
        var op_start_second = convertSecond(op_start)
        var op_end_second = convertSecond(op_end)
        var pos_second = skipOpening(convertSecond(pos), op_start_second, op_end_second)
        
        commands.forEach{
            when(it) {
                "next" -> pos_second = skipOpening(next(pos_second, video_len_second), op_start_second, op_end_second)
                else -> pos_second = skipOpening(prev(pos_second), op_start_second, op_end_second)
            }
        }   
        return convertTimeFormat(pos_second)
    }
    fun next(pos : Int, video_len : Int) : Int {
        return if(pos+10 > video_len) video_len else pos+10
    }
    fun prev(pos : Int) : Int {
        return if(pos-10 < 0) 0 else pos-10
    }
    fun skipOpening(pos: Int, op_start: Int, op_end: Int) : Int {
        return if(pos >= op_start && pos <=op_end) op_end else pos
    }
    fun convertSecond(time : String) : Int {
        var (m,s) = time.split(":")
        return m.toInt()*60+s.toInt()
    }
    fun convertTimeFormat(second : Int) : String {
        var m = second/60
        var mString = if(m<10) "0"+m.toString() else m.toString()
        var s = second%60
        var sString = if(s<10) "0"+s.toString() else s.toString()
        return mString+":"+sString
    }
}